### PR TITLE
Apache Kafka: add 2.7.1 and 2.8.0

### PR DESCRIPTION
**Pull request by polynomial**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128043` at 5bb0c682 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>apacheKafka (apacheKafka_2_8)</li>
    <li>apacheKafka_2_7</li>
  </ul>
</details>

File: nixos/tests/kafka.nix [hunk 1]
```diff
@@ -78,4 +78,6 @@ in with pkgs; {
   kafka_2_4  = makeKafkaTest "kafka_2_4"  apacheKafka_2_4;
   kafka_2_5  = makeKafkaTest "kafka_2_5"  apacheKafka_2_5;
   kafka_2_6  = makeKafkaTest "kafka_2_6"  apacheKafka_2_6;
+  kafka_2_7  = makeKafkaTest "kafka_2_7"  apacheKafka_2_7;
+  kafka_2_8  = makeKafkaTest "kafka_2_8"  apacheKafka_2_8;
```
Comment (SuperSandro2000) [inline nixos/tests/kafka.nix:82]: Can we drop some old versions?

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -61,6 +61,9 @@ mapAliases ({
   apacheKafka_2_1 = throw "kafka 2.1 is no longer supported. Please upgrade to a newer version."; # added 2020-12-21
   apacheKafka_2_2 = throw "kafka 2.2 is no longer supported. Please upgrade to a newer version."; # added 2020-12-21
   apacheKafka_2_3 = throw "kafka 2.3 is no longer supported. Please upgrade to a newer version."; # added 2020-12-21
+  apacheKafka_2_4 = throw "kafka 2.4 is no longer supported. Please upgrade to a newer version."; # added 2021-06-24
+  apacheKafka_2_5 = throw "kafka 2.5 is no longer supported. Please upgrade to a newer version."; # added 2021-06-24
+  apacheKafka_2_6 = throw "kafka 2.6 is no longer supported. Please upgrade to a newer version."; # added 2021-06-24
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:66]: Please delete the old versions.

Comment (polynomial) [reply pkgs/top-level/aliases.nix]: done

File: pkgs/servers/apache-kafka/default.nix [hunk 3]
```diff
@@ -23,6 +23,18 @@ let
       sha256 = "1a2kd4r6f8z7qf886nnq9f350sblzzdi230j2hll7x156888573y";
       jre = jre11;
     };
```
Comment (SuperSandro2000) [inline pkgs/servers/apache-kafka/default.nix:25]: Those should also be deleted.

**Pull request merged**